### PR TITLE
Improve error reporting from failed unit tests.

### PR DESCRIPTION
**Pull request by cmelchior**
This PR fixes the `RunInLooperThread` rule so e.g. test timeouts are not shadowed by cleanup methods failing, which makes debugging very hard. After this the chance of failure is lower, and if it happens both test errors + errors from the `after()` method is reported.

@realm/java

### CODE REVIEW

File: realm/realm-library/src/androidTest/java/io/realm/TestHelper.java [hunk 1]
```diff
@@ -702,4 +706,4 @@ public static InputStream loadJsonFromAssets(Context context, String file) throw
         AssetManager assetManager = context.getAssets();
         return assetManager.open(file);
     }
-}
+}
```
Comment (dalinaum) [inline realm/realm-library/src/androidTest/java/io/realm/TestHelper.java]: It causes the failure as follows:

```
:realm-library:checkstyle[ant:checkstyle] /home/jenkins/jenkins/workspace/java_pr/realm/realm-library/src/androidTest/java/io/realm/TestHelper.java:0: File does not end with a newline.
 FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':realm-library:checkstyle'.
> Checkstyle rule violations were found. See the report at: file:///home/jenkins/jenkins/workspace/java_pr/realm/realm-library/build/reports/checkstyle/checkstyle.html
```

**Comment by dalinaum** [comment]
:+1:

File: realm/realm-library/src/androidTest/java/io/realm/rule/RunInLooperThread.java [hunk 2]
```diff
@@ -108,8 +114,32 @@ public void run() {
...
+                            StringBuilder errorMessage = new StringBuilder();
+                            errorMessage.append("after() threw an error that shadows a test case error");
+                            errorMessage.append('\n');
+                            errorMessage.append("== Test case exception ==\n");
+                            errorMessage.append(testStackTrace.toString());
+                            errorMessage.append('\n');
+                            errorMessage.append("== after() exception ==\n");
+                            errorMessage.append(aftertStackTrace.toString());
+                            fail(errorMessage.toString());
+                        }
```
Comment (beeender) [inline realm/realm-library/src/androidTest/java/io/realm/rule/RunInLooperThread.java]: Does it need to throw `e` if `testException` is not null?

**Comment by cmelchior** [comment]
Covered all corner cases if a RunInLooperThread test fails. @beeender / @realm/java

**Comment by beeender** [comment]
:+1:
